src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureFunction/InvokeFunctionAsyncWrapper.cs

```diff
@@ -179,24 +218,107 @@ public FunctionDetails(dynamic functionContext, IAgent agent)
+                var assembly = loadedAssemblies.FirstOrDefault(a => a.GetName().Name == "Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore");
+
+                _hasAspNetCoreExtensionsReference = assembly != null;
+
+                if (_hasAspNetCoreExtensionsReference.Value)
+                    agent.Logger.Debug("Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore assembly is loaded; InvokeFunctionAsyncWrapper will defer HttpTrigger parameter parsing to FunctionsHttpProxyingMiddlewareWrapper.");
+            }
+
+            // don't parse request parameters here if the Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore assembly is loaded.
+            // If it is loaded, parsing occurs over in FunctionsHttpProxyingMiddlewareWrapper
+            if (_hasAspNetCoreExtensionsReference.Value)
+            {
+                return;
+            }
+
+            object features = functionContext.Features;
+
+            if (_genericFunctionInputBindingFeatureGetter == null) // cache the methodinfo lookups for performance
+            {
+                var get = features.GetType().GetMethod("Get");
+                if (get != null)
+                {
+                    _genericFunctionInputBindingFeatureGetter = get.MakeGenericMethod(features.GetType().Assembly.GetType("Microsoft.Azure.Functions.Worker.Context.Features.IFunctionInputBindingFeature"));
+                }
+                else
+                {
+                    agent.Logger.Debug("Unable to find FunctionContext.Features.Get method; unable to parse request parameters.");
+                    return;
+                }
+
+                var bindFunctionInputType = features.GetType().Assembly.GetType("Microsoft.Azure.Functions.Worker.Context.Features.IFunctionInputBindingFeature");
+                if (bindFunctionInputType == null)
+                {
+                    agent.Logger.Debug("Unable to find IFunctionInputBindingFeature type; unable to parse request parameters.");
+                    return;
+                }
+                _bindFunctionInputAsync = bindFunctionInputType.GetMethod("BindFunctionInputAsync");
+                if (_bindFunctionInputAsync == null)
+                {
+                    agent.Logger.Debug("Unable to find BindFunctionInputAsync method; unable to parse request parameters.");
+                    return;
+                }
+            }
+
+            if (_genericFunctionInputBindingFeatureGetter != null)
+            {
+                // Get the input binding feature and bind the input from the function context
+                var inputBindingFeature = _genericFunctionInputBindingFeatureGetter.Invoke(features, []);
+                dynamic valueTask = _bindFunctionInputAsync.Invoke(inputBindingFeature, [functionContext]);
+
+                valueTask.AsTask().Wait(); // BindFunctionInputAsync returns a ValueTask, so we need to convert it to a Task to wait on it
```

Comment:
Is this really our best option for getting access to the inputs? Feels a bit clunky.

Reply:
This is, unfortunately, exactly what the GRPC-generated code in an Azure function app does between when the `InvokeFunctionAsync()` method is called and when the actual function `Run()` method is called. It's ugly, but it's all we have to work with. 

```
        public async global::System.Threading.Tasks.ValueTask ExecuteAsync(global::Microsoft.Azure.Functions.Worker.FunctionContext context)
        {
            var inputBindingFeature = context.Features.Get<global::Microsoft.Azure.Functions.Worker.Context.Features.IFunctionInputBindingFeature>();
            var inputBindingResult = await inputBindingFeature.BindFunctionInputAsync(context);
            var inputArguments = inputBindingResult.Values;

            if (string.Equals(context.FunctionDefinition.EntryPoint, "AzureFunctionApp.Function1.Run", StringComparison.Ordinal))
            {
                var instanceType = types["AzureFunctionApp.Function1"];
                var i = _functionActivator.CreateInstance(instanceType, context) as global::AzureFunctionApp.Function1;
                context.GetInvocationResult().Value = await i.Run((global::Microsoft.Azure.Functions.Worker.Http.HttpRequestData)inputArguments[0]);
            }
            else if (string.Equals(context.FunctionDefinition.EntryPoint, "AzureFunctionApp.Function2.Run", StringComparison.Ordinal))
            {
                var instanceType = types["AzureFunctionApp.Function2"];
                var i = _functionActivator.CreateInstance(instanceType, context) as global::AzureFunctionApp.Function2;
                context.GetInvocationResult().Value = i.Run((global::Microsoft.AspNetCore.Http.HttpRequest)inputArguments[0]);
            }
            else if (string.Equals(context.FunctionDefinition.EntryPoint, "AzureFunctionApp.QueueTriggerFunction.Run", StringComparison.Ordinal))
            {
                var instanceType = types["AzureFunctionApp.QueueTriggerFunction"];
                var i = _functionActivator.CreateInstance(instanceType, context) as global::AzureFunctionApp.QueueTriggerFunction;
                i.Run((string)inputArguments[0]);
            }
        }
```